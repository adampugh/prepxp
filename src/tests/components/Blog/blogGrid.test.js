import React from "react";
import { shallow } from "enzyme";
import BlogGrid from "../../../components/Blog/blogGrid";
import posts from "../../fixtures/blogPosts";

describe("<blogGrid />", () => {
    test("should match snapshot", () => {
        const wrapper = shallow(<BlogGrid posts={posts} />);
        expect(wrapper).toMatchSnapshot();
    });

    test("should render loading while waiting for posts", () => {
        const wrapper = shallow(<BlogGrid posts={[]} loading={true}/>);
        
    })
});