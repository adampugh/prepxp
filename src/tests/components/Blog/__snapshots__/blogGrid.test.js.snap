// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<blogGrid /> should match snapshot 1`] = `
<div
  className="blog__background"
>
  <div
    className="container"
  >
    <div
      className="blog__grid__container"
    >
      <h1
        className="blog__title"
      >
        Editor's Picks
      </h1>
      <Link
        key="1234"
        replace={false}
        to={
          Object {
            "pathname": "/prepxp/blog/5-Tips-to-get-things-done",
            "state": Object {
              "post": Object {
                "byline": "yesyesyeys",
                "id": "1234",
                "img": "http://via.placeholder.com/300x300",
                "text": "sf sdfg dfasd df dg asdf sdf adsf sadf yea oia the ruler is bakc ofi course is i sfsf aoosd sgodf sfoeg wefodfguonfls sdfsd fsd sgosfsd sdfpsdf sdgxfogfdng  sdfosf sgosf sdfosg sdosf sgosdf sdfosdg sfosdf sd",
                "title": "5 Tips to get things done",
              },
              "posts": Array [
                Object {
                  "byline": "yesyesyeys",
                  "id": "1234",
                  "img": "http://via.placeholder.com/300x300",
                  "text": "sf sdfg dfasd df dg asdf sdf adsf sadf yea oia the ruler is bakc ofi course is i sfsf aoosd sgodf sfoeg wefodfguonfls sdfsd fsd sgosfsd sdfpsdf sdgxfogfdng  sdfosf sgosf sdfosg sdosf sgosdf sdfosdg sfosdf sd",
                  "title": "5 Tips to get things done",
                },
              ],
            },
          }
        }
      >
        <blogPanel
          comments={
            Array [
              Object {
                "name": "Bob",
                "postId": "testPostId",
                "text": "test comment",
                "uid": "cooluid",
              },
            ]
          }
          post={
            Object {
              "byline": "yesyesyeys",
              "id": "1234",
              "img": "http://via.placeholder.com/300x300",
              "text": "sf sdfg dfasd df dg asdf sdf adsf sadf yea oia the ruler is bakc ofi course is i sfsf aoosd sgodf sfoeg wefodfguonfls sdfsd fsd sgosfsd sdfpsdf sdgxfogfdng  sdfosf sgosf sdfosg sdosf sgosdf sdfosdg sfosdf sd",
              "title": "5 Tips to get things done",
            }
          }
        />
      </Link>
      <div
        className="blog__grid"
      >
        <Link
          key="5 Tips to get things done0"
          replace={false}
          to={
            Object {
              "pathname": "/prepxp/blog/5-Tips-to-get-things-done",
              "state": Object {
                "post": Object {
                  "byline": "yesyesyeys",
                  "id": "1234",
                  "img": "http://via.placeholder.com/300x300",
                  "text": "sf sdfg dfasd df dg asdf sdf adsf sadf yea oia the ruler is bakc ofi course is i sfsf aoosd sgodf sfoeg wefodfguonfls sdfsd fsd sgosfsd sdfpsdf sdgxfogfdng  sdfosf sgosf sdfosg sdosf sgosdf sdfosdg sfosdf sd",
                  "title": "5 Tips to get things done",
                },
                "posts": Array [
                  Object {
                    "byline": "yesyesyeys",
                    "id": "1234",
                    "img": "http://via.placeholder.com/300x300",
                    "text": "sf sdfg dfasd df dg asdf sdf adsf sadf yea oia the ruler is bakc ofi course is i sfsf aoosd sgodf sfoeg wefodfguonfls sdfsd fsd sgosfsd sdfpsdf sdgxfogfdng  sdfosf sgosf sdfosg sdosf sgosdf sdfosdg sfosdf sd",
                    "title": "5 Tips to get things done",
                  },
                ],
              },
            }
          }
        >
          <blogPanel
            comments={
              Array [
                Object {
                  "name": "Bob",
                  "postId": "testPostId",
                  "text": "test comment",
                  "uid": "cooluid",
                },
              ]
            }
            post={
              Object {
                "byline": "yesyesyeys",
                "id": "1234",
                "img": "http://via.placeholder.com/300x300",
                "text": "sf sdfg dfasd df dg asdf sdf adsf sadf yea oia the ruler is bakc ofi course is i sfsf aoosd sgodf sfoeg wefodfguonfls sdfsd fsd sgosfsd sdfpsdf sdgxfogfdng  sdfosf sgosf sdfosg sdosf sgosdf sdfosdg sfosdf sd",
                "title": "5 Tips to get things done",
              }
            }
          />
        </Link>
      </div>
      <h1
        className="blog__title"
      >
        Latest Posts
      </h1>
      <div
        className="blog__grid"
      >
        <Link
          key="1234"
          replace={false}
          to={
            Object {
              "pathname": "/prepxp/blog/5-Tips-to-get-things-done",
              "state": Object {
                "post": Object {
                  "byline": "yesyesyeys",
                  "id": "1234",
                  "img": "http://via.placeholder.com/300x300",
                  "text": "sf sdfg dfasd df dg asdf sdf adsf sadf yea oia the ruler is bakc ofi course is i sfsf aoosd sgodf sfoeg wefodfguonfls sdfsd fsd sgosfsd sdfpsdf sdgxfogfdng  sdfosf sgosf sdfosg sdosf sgosdf sdfosdg sfosdf sd",
                  "title": "5 Tips to get things done",
                },
                "posts": Array [
                  Object {
                    "byline": "yesyesyeys",
                    "id": "1234",
                    "img": "http://via.placeholder.com/300x300",
                    "text": "sf sdfg dfasd df dg asdf sdf adsf sadf yea oia the ruler is bakc ofi course is i sfsf aoosd sgodf sfoeg wefodfguonfls sdfsd fsd sgosfsd sdfpsdf sdgxfogfdng  sdfosf sgosf sdfosg sdosf sgosdf sdfosdg sfosdf sd",
                    "title": "5 Tips to get things done",
                  },
                ],
              },
            }
          }
        >
          <blogPanel
            comments={
              Array [
                Object {
                  "name": "Bob",
                  "postId": "testPostId",
                  "text": "test comment",
                  "uid": "cooluid",
                },
              ]
            }
            post={
              Object {
                "byline": "yesyesyeys",
                "id": "1234",
                "img": "http://via.placeholder.com/300x300",
                "text": "sf sdfg dfasd df dg asdf sdf adsf sadf yea oia the ruler is bakc ofi course is i sfsf aoosd sgodf sfoeg wefodfguonfls sdfsd fsd sgosfsd sdfpsdf sdgxfogfdng  sdfosf sgosf sdfosg sdosf sgosdf sdfosdg sfosdf sd",
                "title": "5 Tips to get things done",
              }
            }
          />
        </Link>
      </div>
    </div>
  </div>
</div>
`;

exports[`<blogGrid /> should render error message if error 1`] = `
<div
  className="blog__background"
>
  <div
    className="container"
  >
    <h1
      className="blog__grid__error"
    >
      Sorry, an error has occured
    </h1>
  </div>
</div>
`;

exports[`<blogGrid /> should render loading while waiting for posts 1`] = `
<div
  className="blog__background"
>
  <div
    className="container"
  >
    <loader />
  </div>
</div>
`;
